### DISCUSSION TITLE
Return values from a child dag triggered without execution date using TriggerDagRunOperator

### DISCUSSION THREAD

**Discussion by sachinmn2010** [apache/airflow#17205] [Q&A] [answered]
**Context:**
I have a set of operators in a dag that perform operations based on the dag_run configuration(let's call it child dag: dag_c). The child dag can be triggered by other dags (let's call it Parent dag_a & dag_b). Parent dag_a & dag_b runs at the same time. 

**Use case:** 
My use case is to get a return value from dag c to dag a and dag b. The return value is unique to the parent dag which triggered the child dag. Since dag A & B can run at the same time, I cannot use execution_date. The code in dag c is common for both dag a & b, and I don't want to copy-paste the same code in both dags. 

**Problem Statement:** 
Without setting the execution date, I am not able to get the return value using the xcom_pull method

So far I have tried :
1. To push the return value using the xcom_push method in child dag, but the child dags execution is in future and value is not available in parent dag 
2. Tried to do an xcom_push using the parent dag's execution, but it failed since we can not set an x_com in the past. The job fails with error: 'execution_date can not be in the past'
3. Use Subdags for dag c generation and call it in dag A & B. This method solved my use case, but sub dags are not recommended in [airflow](https://www.astronomer.io/guides/subdags), I would like to know if there are any better methods to solve this problem.

I am using airflow 2.0.2.

**Comment by josh-fell** [comment, marked as answer]
Maybe try Airflow Variables instead of XCom in this case. You could use the Variable.set() method to write the return value required. There would not be any execution_date constraints on the value that's set and the value is still centrally accessible by dag_a and dag_b.

**Reply by potiuk** [reply]
My thought exactly. I am not sure if this is the right approach to have multiple DAGs in this - case, there are other options - TaskGroups, SubDags, but if you want to use multiple DAGs, Variables should work

**Reply by sachinmn2010** [reply]
thanks, @josh-fell & @potiuk. Using variables makes sense
